static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.spec.js

```diff
@@ -247,50 +304,78 @@ describe('Membership engagement banner', () => {
                 linkUrl: 'fake-link-url',
                 buttonCaption: 'fake-button-caption',
             }));
-            fakeConfig.get.mockImplementationOnce(
-                () => 'fake-paypal-and-credit-card-image'
-            );
+            fakeConfig.get
+                .mockImplementationOnce(() => true)
+                .mockImplementationOnce(
+                    () => 'fake-paypal-and-credit-card-image'
+                );
             fakeInlineSvg.mockImplementationOnce(() => 'fake-button-svg');
             fakeConstructQuery.mockImplementationOnce(
                 () => 'fake-query-parameters'
             );
-            showBanner = membershipEngagementBannerInit().then(() => {
-                fakeMediator.emit('modules:onwards:breaking-news:ready', false);
-            });
         });
 
         it('message text', () =>
-            showBanner.then(() => {
+            membershipEngagementBanner.canShow().then(canShow => {
+                expect(canShow).toBe(true);
+
+                membershipEngagementBanner.show();
+
                 expect(FakeMessage.prototype.show.mock.calls[0][0]).toMatch(
                     /fake-message-text/
                 );
             }));
+
         it('paypal and credit card image', () =>
-            showBanner.then(() => {
+            membershipEngagementBanner.canShow().then(canShow => {
+                expect(canShow).toBe(true);
+
+                membershipEngagementBanner.show();
+
                 expect(FakeMessage.prototype.show.mock.calls[0][0]).toMatch(
                     /fake-paypal-and-credit-card-image/
                 );
             }));
+
         it('colour class', () =>
-            showBanner.then(() => {
+            membershipEngagementBanner.canShow().then(canShow => {
+                expect(canShow).toBe(true);
+
+                membershipEngagementBanner.show();
+
                 expect(FakeMessage.prototype.show.mock.calls[0][0]).toMatch(
                     /fake-colour-class/
                 );
             }));
+
         it('link URL', () =>
-            showBanner.then(() => {
+            membershipEngagementBanner.canShow().then(canShow => {
+                expect(canShow).toBe(true);
+
+                membershipEngagementBanner.show();
+
                 expect(FakeMessage.prototype.show.mock.calls[0][0]).toMatch(
                     /fake-link-url\?fake-query-parameters/
                 );
             }));
+
         it('button caption', () =>
-            showBanner.then(() => {
+            membershipEngagementBanner.canShow().then(canShow => {
+                expect(canShow).toBe(true);
```

Comment:
Personally i would call `show()` directly from this set of tests (since you are already testing `canShow()` outside). Should shave a couple milliseconds off the overall testing.

```
it('button SVG', () => {
  membershipEngagementBanner.show();
  return expect(FakeMessage.prototype.show.mock.calls[0][0]).toMatch(
    /fake-button-svg/
  )
});
```

Reply:
Agreed, If I was writing the tests from scratch I would split them into a set of tests calling `show` directly and `canShow` directly, however the approach taken was the quickest way to repurpose the tests that were already in place.

Reply:
What I meant is that there's already a bunch of tests for canShow() you converted up so there's really no point in even calling it in this area is it? you could just bypass that entire bit

Reply:
`bannerParams` is assigned a value in `canShow` and is subsequently used in `show`/`showBanner`, so I'd have to look at setting up a way of defining `bannerParams` without calling `canShow` from the test suite.

Reply:
aughhhh this one too? ToT